File: app/src/main/java/com/example/taxconnect/model/BookingModel.java
package com.example.taxconnect.model;

import java.io.Serializable;

public class BookingModel implements Serializable {
    private String id;
    private String caId;
    private String userId;
    private String userName; // User's name for CA to see
    private String caName; // CA's name for User to see
    private long appointmentTimestamp;
    private String status; // PENDING, CONFIRMED, COMPLETED, CANCELLED
    private long createdAt;
    private String message;

    public BookingModel() {}

    public BookingModel(String id, String caId, String userId, String userName, String caName, long appointmentTimestamp, String status, String message) {
        this.id = id;
        this.caId = caId;
        this.userId = userId;
        this.userName = userName;
        this.caName = caName;
        this.appointmentTimestamp = appointmentTimestamp;
        this.status = status;
        this.message = message;
        this.createdAt = System.currentTimeMillis();
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getCaId() { return caId; }
    public void setCaId(String caId) { this.caId = caId; }

    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public String getUserName() { return userName; }
    public void setUserName(String userName) { this.userName = userName; }

    public String getCaName() { return caName; }
    public void setCaName(String caName) { this.caName = caName; }

    public long getAppointmentTimestamp() { return appointmentTimestamp; }
    public void setAppointmentTimestamp(long appointmentTimestamp) { this.appointmentTimestamp = appointmentTimestamp; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public long getCreatedAt() { return createdAt; }
    public void setCreatedAt(long createdAt) { this.createdAt = createdAt; }
}

File: tax connect/app/src/main/java/com/example/taxconnect/data/models/BookingModel.kt
package com.example.taxconnect.data.models

import java.io.Serializable

data class BookingModel(
    var id: String? = null,
    var caId: String? = null,
    var userId: String? = null,
    var userName: String? = null,         // User's name for CA to see
    var caName: String? = null,           // CA's name for User to see
    var serviceId: String? = null,
    var serviceName: String? = null,      // Selected service label
    var appointmentTimestamp: Long = 0,
    var appointmentDate: String? = null,  // Human-readable date string, e.g. "21 Mar 2026"
    var appointmentTime: String? = null,  // Human-readable time string, e.g. "10:00 AM"
    var status: String? = null,           // PENDING, ACCEPTED, COMPLETED, CANCELLED, REJECTED, EXPIRED
    var chatId: String? = null,           // Set after CA accepts — links to conversation
    var createdAt: Long = System.currentTimeMillis(),
    var updatedAt: Long = 0,
    var message: String? = null,
    var rejectionReason: String? = null,  // Reason provided by CA when rejecting
    var appointmentHour: Int = -1,        // 24-hour component of the slot (0-23); -1 = not set
    var appointmentMinute: Int = 0,       // Minute component of the slot
    var totalAmount: Double = 0.0,
    var advanceAmount: Double = 0.0
) : Serializable
